packages/remediations/src/utils/utils.js

```diff
@@ -177,18 +179,45 @@ const changeBulkSelect = (state, action) => {
     });
 };
 
-export const fetchSystemsInfo = async (config, allSystems, { getEntities } = {}) => {
-    const systems = (allSystems || []).slice((config.page - 1) * config.per_page, config.page * config.per_page);
-    const data = await getEntities(systems, { ...config, hasItems: true, page: 1 }, true);
+export const fetchSystemsInfo = async (config, allSystemsNamed = [], { getEntities } = {}) => {
+    const hostnameOrId = config?.filters?.hostnameOrId?.toLowerCase();
+    const systems = (
+        hostnameOrId
+            ?
+            allSystemsNamed.reduce((acc, curr) => [
+                ...acc,
+                ...(curr.display_name.toLowerCase().includes(hostnameOrId) || curr.id.toLowerCase().includes(hostnameOrId)
+                    ? [ curr.id ]
+                    : []
+                )
+            ], [])
+            : allSystemsNamed.map(system => system.id)
+    );
+    const sliced = systems.slice((config.page - 1) * config.per_page, config.page * config.per_page);
+    const data = sliced.length > 0 ? await getEntities(sliced, { ...config, hasItems: true, page: 1 }, true) : {};
     return {
         ...data,
-        total: allSystems.length,
+        total: systems.length,
         page: config.page,
-        // eslint-disable-next-line camelcase
         per_page: config.per_page
     };
 };
 
+export const splitArray = (inputArray, perChunk) => inputArray.reduce((resultArray, item, index) => {
+    const chunkIndex = Math.floor(index / perChunk);
+    if (!resultArray[chunkIndex]) {
+        resultArray[chunkIndex] = [];
+    }
+
+    resultArray[chunkIndex].push(item);
+    return resultArray;
+}, []);
```

Comment:
This is too complex, you can achieve the same result by using `slice`, you are making too much iterations
```JS
export const splitArray = [...new Array(Math.ceil(inputArray.length/perChunk))].map((_item, key) => inputArray.slice(key * perChunk, (key + 1) * perChunk))
```

You can test it on https://jsben.ch/cD5xN, the results are quite impressive. The sollution with `split` is almost twice as fast for medium arrays. For small ones the difference is just a few percentage, but once we enter the large sized arrays the difference is huge (20 % for the one in this PR).

Reply:
@karelhala wow, really cool! Thank you :)